src/2d/spreadinterp2d.cu

```diff
@@ -529,15 +528,21 @@ void Interp_2d_NUptsdriven(FLT *x, FLT *y, CUCPX *c, CUCPX *fw, int M, const int
 		CUCPX cnow;
 		cnow.x = 0.0;
 		cnow.y = 0.0;
+		FLT ker1[MAX_NSPREAD];
+		FLT ker2[MAX_NSPREAD];
+
+		FLT x1=(FLT)xstart-x_rescaled;
+		FLT y1=(FLT)ystart-y_rescaled;
+		eval_kernel_vec(ker1,x1,ns,es_c,es_beta);
+		eval_kernel_vec(ker2,y1,ns,es_c,es_beta);
```

Comment:
in FINUFFT we combine the x and y direction kernels into a single call - might be more efficient (SIMD? not sure matters on GPU).  All allocations are then length 3*MAX_NSPREAD just for convenience in all dims.